src/GapicClientTrait.php

```diff
@@ -238,6 +238,12 @@ private function setClientOptions(array $options)
             'libName',
             'libVersion',
         ]);
+
+        // "hasEmulator" is not a supported Client Option, but is used
+        // internally to determine if the client is running in emulator mode.
+        // Therefore, we need to remove it from the $options array before
+        // creating the ClientOptions.
+        $hasEmulator = (bool) $this->pluck('hasEmulator', $options, false);
```

Comment:
Is this casting problematic? I see that `pluck` has a `false` parameter, I am assuming that this is a default return value if is not found?

Would this not return a value nor throw an exception in any other cases?

Reply:
the `false` is (not intuitively) to say that the parameter isn't required. Without that, it would throw an exception if the key is not present in the array.

This parameter is only to be set by us, so it isn't super important to cast it. But the `bool` cast would never throw a type error, so if someone passed an object, stream, etc into `hasEmulator`, it would result in a `bool`. 

```php
var_dump((bool) new stdClass());
// bool(true)
function foo(bool $bar)
{
    var_dump($bar);
}

foo('123');
// bool(true)
foo(new stdClass());
// Uncaught TypeError: foo(): Argument #1 ($bar) must be of type bool, stdClass given
```

I could see how a type error would be preferred, so we could remove this cast so the typehinting is enforced:

```suggestion
        $hasEmulator = $this->pluck('hasEmulator', $options, false);
```